src/components/ConnectionsList.tsx

```diff
@@ -0,0 +1,149 @@
+  );
+  const [selectedListTitle, setSelectedListTitle] = useState<ListStatus>(
+    ListStatus.CLOSED
+  );
+  const [selectedList, setSelectedList] = useState<Profile[]>([]);
+  const [followingList, setFollowingList] = useState<Profile[]>([]);
+  const [followersList, setFollowersList] = useState<Profile[]>([]);
+  const [notFollowing, setNotFollowing] = useState<Profile[]>([]);
+  const dispatch = useAppDispatch();
+
+  const getConnectionProfile = async (
+    socialAddress: HexString
+  ): Promise<Profile> => {
+    let userProfile = cachedProfiles.find(
+      (profile) => profile.socialAddress === socialAddress
+    );
+    if (!userProfile) {
+      userProfile = await sdk.getProfile(socialAddress);
+      dispatch(upsertProfile(userProfile));
+    }
+    return userProfile;
+  };
+
+  const getUserConnectionsList = async (
+    following: HexString[],
+    followers: HexString[]
+  ) => {
+    const followingProfiles: Profile[] = await Promise.all(
+      (following || []).map(
+        async (socialAddress: string) =>
+          await getConnectionProfile(socialAddress)
+      )
+    );
+    const followersProfiles: Profile[] = await Promise.all(
+      (followers || []).map(
+        async (socialAddress: string) =>
+          await getConnectionProfile(socialAddress)
+      )
+    );
+    return [followingProfiles, followersProfiles];
+  };
+
+  const getNotFollowingProfiles = (
+    followingProfiles: Profile[],
+    followersProfiles: Profile[]
+  ) => {
+    return followersProfiles.filter((userProfile) => {
+      return !followingProfiles.find(
+        (followingProfile) =>
+          followingProfile.socialAddress === userProfile.socialAddress
+      );
+    });
+  };
+
+  useEffect(() => {
+    if (!graph) return;
+    getUserConnectionsList(graph.following, graph.followers).then(
+      (userRelationships) => {
+        const [followingProfiles, followersProfiles] = userRelationships;
+        setFollowingList(followingProfiles);
+        setFollowersList(followersProfiles);
+        setNotFollowing(
+          getNotFollowingProfiles(followingProfiles, followersProfiles)
+        );
+      }
+    );
+  }, [graph]);
+
+  useEffect(() => {
+    if (selectedListTitle === ListStatus.FOLLOWING) {
+      setSelectedList(followingList);
+    } else if (selectedListTitle === ListStatus.FOLLOWERS) {
+      setSelectedList(followersList);
+    } else setSelectedList([]);
+  }, [selectedListTitle]);
+
+  const handleClick = (listTitle: ListStatus) => {
+    if (selectedListTitle === listTitle)
+      setSelectedListTitle(ListStatus.CLOSED);
+    else {
+      setSelectedListTitle(listTitle);
+    }
+  };
+
+  const getClassName = (name: ListStatus) => {
+    return selectedListTitle === name
+      ? "ConnectionsList__button ConnectionsList__button--active"
+      : "ConnectionsList__button";
+  };
+
+  return (
+    <div className="ConnectionsList__block">
+      <div className="ConnectionsList__buttonBlock">
+        <Button
```

Comment:
```
    <Button
          className={getClassName(ListStatus.FOLLOWERS)}
          onClick={() => handleClick(ListStatus.FOLLOWERS)}
        >
          <div className="ConnectionsList__buttonCount">
            {graph?.followers.length}
          </div>
          Followers
        </Button>
        ```
        maybe you can make this a small component

Reply:
I kind of like having it right here so you.  more readable instead of having to dig and figure out where everything is.

Reply:
you can keep it in the same file